src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -767,6 +767,28 @@ private void RegisterCustomNodeDefinitionWithEngine(CustomNodeDefinition definit
                 Workspaces.OfType<HomeWorkspaceModel>().SelectMany(ws => ws.Nodes),
                 definition,
                 DebugSettings.VerboseLogging);
+
+            MarkAllDependenciesAsModified(definition);
+        }
+
+        /// <summary>
+        /// Get all function instances or directly or indrectly dependo on the 
+        /// specified function definition and mark them as modified so that 
+        /// their values will be re-queryed.
+        /// </summary>
+        /// <param name="functionId"></param>
+        /// <returns></returns>
+        private void MarkAllDependenciesAsModified(CustomNodeDefinition def)
+        {
+            var homeWorkspace = Workspaces.OfType<HomeWorkspaceModel>().FirstOrDefault();
```

Comment:
`homeWorkspace` may be `null` simply because we call `FirstOrDefault`, I think it's best to have a `null` check here.